src/spaceObjects/spaceshipParts/weaponTube.cpp

```diff
@@ -284,6 +285,13 @@ bool WeaponTube::isUnloading()
     return state == WTS_Unloading;
 }
 
+bool WeaponTube::hasFired()
+{
+    bool flag = has_fired;
+    if (has_fired) has_fired = false;
```

Comment:
It's very implicit behavour that a "hasFired" function resets the flag as well. Not saying I have a better solution, but it's not clear from the function name that calling it will change internal state of the weapon tube.